### ISSUE TITLE
PEP 541 Request: hype

### ISSUE THREAD

**Issue by mattt** [pypi/support#4757]
### Project to be claimed

`PROJECT_NAME`: https://pypi.org/project/hype

### Your PyPI username

`USER_NAME`: https://pypi.org/user/mattt

### Reasons for the request

I’m requesting ownership of the “hype” package on PyPI. Based on PEP 541, this package appears to be abandoned and potentially namesquatted. Here’s why:

1. No updates for over 10 Years: The only release of “hype” was on November 6, 2013—over 10 years ago. Since then, there have been no updates or activity, suggesting it’s abandoned.
2. Empty package: The package is empty and doesn’t provide any functionality, indicating it may have been uploaded just to reserve the name without any real development.
3. Inaccessible project source: Navigating to the homepage / GitHub repository linked results in a 404 error.
4. Plan for active maintenance: I have a [new project](https://github.com/mattt/hype) for which I'd like to use the package name "hype" and plan to actively develop and maintain.

### Maintenance or replacement?

Replacement

### Source code repositories URLs

Current: https://github.com/balanced/hype (404)

Replacement: https://github.com/mattt/hype

### Contact and additional research

I reached out to the listed maintainer for the project, Mahmoud Abdelkader, using the email listed on their GitHub profile (https://github.com/mahmoudimus) on September 10, 2024 and followed up on September 13, 2024, but have yet to receive a response.

### Code of Conduct

- [X] I agree to follow the PSF Code of Conduct

**Comment by mattt** [comment]
Update: I emailed the project maintainer again on October 4th and reached out on Twitter, but have still not received a response from them.

Given that I've been unable to contact the author after 5 weeks, I believe that this issue will only be resolved with the help of a PyPI admin. This naming issue is the last thing blocking me from releasing my project, so I'd greatly appreciate anything y'all can do to help.

**Comment by Thespi-Brain** [comment]
Hi @mattt,

We are working to contact the owner to determine if they're reachable and have sent them a courtesy message.

Final determination regarding project ownership will be made by the PyPI Administrators when the research process is complete.

**Comment by mattt** [comment]
Thanks for the update, @Thespi-Brain — I really appreciate it.

**Comment by Thespi-Brain** [comment]
**hype** has been deemed invalid. Additionally, the owner did not respond back regarding the status of their project. 

I recommend that the PyPI Admins release the name for future use.

**Comment by Thespi-Brain** [comment]
Plot twist: the owner replied back expressing amazement that their package is empty and promised to update it. They have updated it with some code, so even though it's technically no longer empty, changing it's status to "Special cases." Up to the discretion of PyPI admins on whether or not this still constitutes some form of name squat and if the package should be released or more updates are needed from the package owner.

**Comment by mattt** [comment]
@Thespi-Brain Thanks for the update. That is indeed an interesting twist.

I downloaded the [source files for this 1.1 release of that package](https://pypi.org/project/hype/#files). Here's what I found:

```console
❯ tree ./hype-1.1 
./hype-1.1
├── PKG-INFO
├── README.rst
├── hype.egg-info
│   ├── PKG-INFO
│   ├── SOURCES.txt
│   ├── dependency_links.txt
│   └── top_level.txt
├── setup.cfg
└── setup.py

The package contains no Python code, aside from a `setup.py` file containing metadata. The metadata declares compatibility with Python 2.7, which is consistent with the original date of publication of hype in 2013.

The only file with any content is a README. Some other observations about the README:

- The referenced [GitHub repo](https://github.com/balanced/hype/) resolves with a 404. 
- The referenced company, Balanced Payments, appears to have [shut down in 2015](https://news.ycombinator.com/item?id=9200017); its [domain](https://balancedpayments.com) is parked.

Most notably, the README contains the following example code:

```python
from hype import Resource, fields

class User(Resource):
    name = fields.String()
    email = fields.String()
    profile = fields.Link(endpoint='profile.show')

# Transform to JSON
from hype.mime import application_json

def url_for(ref):
    return f'/api/{ref.name}/{ref.params["id"]}'

user = User(name='John Doe', email='john@example.com')
json_output = application_json.serialize(user, url_for=url_for)
```

The `url_for` method uses an f-string, a feature [added in Python 3.6](https://docs.python.org/3/whatsnew/3.6.html). [PEP 498](https://peps.python.org/pep-0498/) was created August 1, 2015 — 5 months after [Balanced Payments announced it was shutting down](https://news.ycombinator.com/item?id=9200017).

Based on this, I don't believe this code is consistent with an existing project that sat empty since 2013 by accident. I hope that the PyPI admins will weigh the facts of the case and follow the guidelines set forth in PEP 541 to resolve this matter.

Thank you again for looking into this, @Thespi-Brain.

**Comment by ewdurbin** [comment]
I agree that the non-functioning project uploaded does not constitute any more valid use of the name on PyPI. Added 'mattt' as 'Owner' on 'hype'
